### PR TITLE
fix(local-runtime): degrade MTP posture after grown window restore

### PR DESCRIPTION
**Pull request by KoNit-K**
## What does this PR do?

Managed llama.cpp picks stacked MTP prefill (`-b 4096 -ub 2048`) from the *initial* window via `_choose_mtp_posture`. After `_restore_grown_window` lifts ctx from `window_overrides.json`, that stacked logits overhead can force `--override-tensor` CPU spill even when the same grown window still fits under lean MTP.

This PR adds a post-restore re-check in `_preset_for`: if stacked spills at the restored window but lean does not, degrade to lean (keep the grown window; do not shrink it). Fail-open when lean also spills — keep stacked and report spill honestly.

Partial fix for #106895 (§3 only). Does **not** add spill UI warnings or a persistent `extra_args` / user preset layer (§1/§2).

## Related Issue

Addresses #106895

## Type of Change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✅ Tests

## Changes Made

- `hermes_cli/local_runtime/presets.py`: `_degrade_mtp_if_grown_spills` + call site after grown-window restore
- `tests/hermes_cli/test_local_growth.py`: RED/GREEN for degrade path + fail-open both-spill CONTROL

## How to Test

```
HERMES_PYTHON=<venv-with-pytest> bash scripts/run_tests.sh tests/hermes_cli/test_local_growth.py -q
# -> 11 passed
```

Proof on base `ecfcbb6276` without fix: `test_mtp_degrades_to_lean_when_grown_window_would_spill` failed with `grown.spilled=True` and `override-tensor` present at window 73728. After fix: same suite green.

## Claim scope / residual risk

- Only §3 (MTP posture re-check). Silent spill still has no chat/desktop surfacing; presets.ini remains fully regenerated (no user override layer).
- UMA inherits existing `launch_args` behavior (no `-ot` on UMA); degrade still drops stacked ubatch when lean fits.

## Checklist

- [x] Focused to the bug seam; no unrelated refactor
- [x] Searched open PRs for #106895 — none at submit time
- [x] Focused tests added and green
- [x] Self-review P0 = 0 (Detection / Fail-open / Forbidden / RED evidence / Diff scope)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by Enough1122** [comment]
> AI code review — automated review for reference; please use your judgment.

## PR 106897 — fix(local-runtime): degrade MTP posture after grown window restore

Correct pricing fix (`presets.py:12-31,43-45`): MTP posture was chosen against the initial window, but a grown override can push stacked-prefill logits into spill while lean still fits. Now, only when the restored window exceeds the initial one, stacked is re-priced and degraded to lean with the window intact and a reason appended; if both postures spill it fails open (keeps stacked, honest spill, no invented shrink).

- Pinned deterministic numbers (16 GiB MoE / 24 GiB card / 151936 vocab with control + both-spill cases) make the posture boundary testable rather than hand-waved. Launch flags and pricing stay consistent via the shared recompute.
- Non-blocking: none functional. The `decision.window > launch_window` gate correctly skips the re-check when an override shrinks the window (stacked still fits a smaller window).

**Comment by KoNit-K** [comment]
Thanks for the review — agreed, no functional blockers.

Confirming the notes as intentional, not follow-ups:

- `_choose_mtp_posture` still prices against the *initial* window; `_degrade_mtp_if_grown_spills` only runs after `_restore_grown_window` when `decision.window > launch_window`. Stacked is re-priced at the restored ctx; if stacked spills and lean fits, we drop to lean and keep the grown window (no invented shrink).
- Both-spill fail-open keeps stacked and reports spill honestly (`override-tensor` stays). Pinned by `test_mtp_keeps_stacked_when_grown_window_still_spills_under_lean`.
- The `decision.window > launch_window` gate correctly skips the re-check when the override does not grow the window (a smaller window still fits stacked). `_restore_grown_window` itself only lifts, never shrinks.
- Launch flags and pricing stay on the same posture via the shared recompute into `launch_args`.

Re-checked against head `92929fb9ef`; focused suite `tests/hermes_cli/test_local_growth.py` 11 passed. No code change from this review.

**Comment by jquesnelle** [comment]
Thanks @KoNit-K for identifying and fixing the restored-window MTP problem. #107959 incorporates that work into a shared effective-window planner covering initial launch, growth and restart, and credits you as a co-author.

The disposition for this PR is **close as superseded by #107959**, consolidating review there now rather than retaining parallel implementations until merge. The replacement is still open; this is not a claim that the fix has shipped. Issue #106895 remains open for spill notifications and persistent parameter controls.
